src/main/java/io/nats/client/impl/NatsConnection.java

```diff
@@ -1472,6 +1472,20 @@ void processError(String errorText) {
         }
     }
 
+    interface ErrorListenerCaller {
+        void call(Connection conn, ErrorListener el);
+    }
+
+    void executeCallback(ErrorListenerCaller elc) {
+        if (!this.callbackRunner.isShutdown()) {
+            try {
+                this.callbackRunner.execute(() -> elc.call(this, options.getErrorListener()));
+            } catch (RejectedExecutionException re) {
+                // Timing with shutdown, let it go
+            }
+        }
+    }
+

```

Comment:
This code allows me to use the existing callbackRunner to fire manager error events. Before it would do in in a synchronous fashion, which could lead to blocking or unhandled runtime exceptions.